spec/rubocop/cop/rails/where_missing_spec.rb

```diff
@@ -0,0 +1,207 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Rails::WhereMissing, :config do
+  context 'Rails 6.1', :rails61 do
+    it 'registers an offense when using `left_joins(foo).where(foos: {id: nil})`' do
+      expect_offense(<<~RUBY)
+        Foo.left_joins(:foo).where(foos: { id: nil }).where(bar: "bar")
+            ^^^^^^^^^^^^^^^^ Use `where.missing(:foo)` instead of `left_joins(:foo).where(foos: { id: nil })`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        Foo.where.missing(:foo).where(bar: "bar")
+      RUBY
+    end
+
+    it 'registers an offense when using `left_outer_joins(foo).where(foos: {id: nil})`' do
+      expect_offense(<<~RUBY)
+        Foo.left_outer_joins(:foo).where(foos: { id: nil }).where(bar: "bar")
+            ^^^^^^^^^^^^^^^^^^^^^^ Use `where.missing(:foo)` instead of `left_outer_joins(:foo).where(foos: { id: nil })`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        Foo.where.missing(:foo).where(bar: "bar")
+      RUBY
+    end
+
+    it 'registers an offense when using `where(foos: {id: nil}).left_joins(foo)`' do
+      expect_offense(<<~RUBY)
+        Foo.where(foos: { id: nil }).left_joins(:foo).where(bar: "bar")
+                                     ^^^^^^^^^^^^^^^^ Use `where.missing(:foo)` instead of `left_joins(:foo).where(foos: { id: nil })`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        Foo.where.missing(:foo).where(bar: "bar")
+      RUBY
+    end
+
+    it "registers an offense when using `left_joins(foo).where(foos: {id: nil}, bar: 'bar')`" do
+      expect_offense(<<~RUBY)
+        Foo.left_joins(:foo).where(foos: { id: nil }, bar: "bar")
+            ^^^^^^^^^^^^^^^^ Use `where.missing(:foo)` instead of `left_joins(:foo).where(foos: { id: nil })`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        Foo.where.missing(:foo).where(bar: "bar")
+      RUBY
+    end
+
+    it "registers an offense when using `left_joins(foo).where(bar: 'bar', foos: {id: nil})`" do
+      expect_offense(<<~RUBY)
+        Foo.left_joins(:foo).where(bar: "bar", foos: { id: nil })
+            ^^^^^^^^^^^^^^^^ Use `where.missing(:foo)` instead of `left_joins(:foo).where(foos: { id: nil })`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        Foo.where.missing(:foo).where(bar: "bar")
+      RUBY
+    end
+
+    it 'registers an offense when using `where(foos: {id: nil}).joins(:bar).left_joins(foo)`' do
+      expect_offense(<<~RUBY)
+        Foo.left_joins(:foo).joins(:bar).where(foos: { id: nil })
+            ^^^^^^^^^^^^^^^^ Use `where.missing(:foo)` instead of `left_joins(:foo).where(foos: { id: nil })`.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        Foo.where.missing(:foo).joins(:bar)
+      RUBY
+    end
+
+    it 'registers an offense when using `left_joins(foo).where(foos: {id: nil})` with multi-line leading ' \
+       'dot method calls' do
+      expect_offense(<<~RUBY)
```

Comment:
Can we also have a test with a multiline where, that also has other conditions, like
```
Foo
           .left_joins(:foo)
           .where(
             foos: { id: nil },
             bar: "bar"
           )
```

Reply:
Thank you so much. I added the test case: https://github.com/rubocop/rubocop-rails/blob/fe2ecf8a8a3fa5975e7201e066402a9ec5f690a9/spec/rubocop/cop/rails/where_missing_spec.rb#L126-L145